concent_api/gatekeeper/views.py

```diff
@@ -124,73 +161,193 @@ def parse_headers(request: WSGIRequest, path_to_file: str) -> Union[FileTransfer
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     assert set(HashingAlgorithm) == {HashingAlgorithm.SHA1}, "If you add a new hashing algorithms, you need to add validations below."
     if any(VALID_SHA1_HASH_REGEX.fullmatch(file_checksum[1]) is None for file_checksum in file_checksums):
-        return gatekeeper_access_denied_response("Invalid SHA1 hash.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "Invalid SHA1 hash.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_INVALID_SHA1_HASH.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     file_sizes = [file["size"] for file in loaded_golem_message.files]
     if any((file_size is None) for file_size in file_sizes):
-        return gatekeeper_access_denied_response("'size' must be an integer.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'size' must be an integer.",
+            ErrorCode.MESSAGE_FILES_SIZE_WRONG_TYPE.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
     for file_size in file_sizes:
         try:
             int(file_size)
         except (ValueError, TypeError):
-            return gatekeeper_access_denied_response("'size' must be an integer.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+            return gatekeeper_access_denied_response(
+                "'size' must be an integer.",
+                ErrorCode.MESSAGE_FILES_SIZE_WRONG_TYPE.value,
+                path_to_file,
+                loaded_golem_message.subtask_id,
+                client_public_key,
+            )
 
     if any(int(file["size"]) < 0  for file in loaded_golem_message.files):
-        return gatekeeper_access_denied_response("'size' must not be negative.", path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            "'size' must not be negative.",
+            ErrorCode.MESSAGE_FILES_SIZE_NEGATIVE.value,
+            path_to_file,
+            loaded_golem_message.subtask_id,
+            client_public_key,
+        )
 
     matching_files = [file for file in loaded_golem_message.files if path_to_file == file['path']]
 
     if len(matching_files) == 1:
         return matching_files[0]
     else:
         assert len(matching_files) == 0
-        return gatekeeper_access_denied_response('Path to specified file is not listed in files variable.', path_to_file, loaded_golem_message.subtask_id, client_public_key)
+        return gatekeeper_access_denied_response(
+            'Path to specified file is not listed in files variable.',
+            ErrorCode.MESSAGE_FILES_PATH_NOT_LISTED_IN_FILES.value,
```

Comment:
`Path to specified file is not listed in files variable.` -> `Your token does not authorize you to transfer the requested file.`

Reply:
:+1: